src/pipelinedefinition.cpp

```diff
@@ -675,12 +677,29 @@ Status PipelineDefinition::validateNodes(ModelManager& manager) {
             return StatusCode::PIPELINE_NODE_NAME_DUPLICATE;
         }
 
+        if (node.demultiplyCount) {
+            isDemultiplexer = true;
+        }
+
+        if (modelConfigs.find(node.modelName) != modelConfigs.end()) {
+            for (auto& [inputName, shapeInfo] : modelConfigs.at(node.modelName).getShapes()) {
+                if (!shapeInfo.shape.empty() && shapeInfo.shape[0] >= 2) {
+                    isMultipleBatchSize = true;
```

Comment:
There are separate methods in ModelInstance for fetching actual batch size and shape. Consider using these. It would also be good idea to use already existing methods for batch size or create one if doesnt exist. (in ModelInstance or ModellConfig)